ovos_dinkum_listener/service.py

```diff
@@ -150,29 +149,58 @@ def __init__(self, on_ready=on_ready, on_error=on_error,
         self.stt = load_stt_module()
         self.fallback_stt = load_fallback_stt()
         self.transformers = AudioTransformersService(self.bus, self.config)
-        self.voice_loop = DinkumVoiceLoop(
-            mic=self.mic,
-            hotwords=self.hotwords,
-            stt=self.stt,
-            fallback_stt=self.fallback_stt,
-            vad=self.vad,
-            transformers=self.transformers,
-            #
-            speech_seconds=listener.get("speech_begin", 0.3),
-            silence_seconds=listener.get("silence_end", 0.7),
-            timeout_seconds=listener.get("recording_timeout", 10),
-            num_stt_rewind_chunks=listener.get("utterance_chunks_to_rewind", 2),
-            num_hotword_keep_chunks=listener.get("wakeword_chunks_to_save", 15),
-            #
-            wake_callback=self._record_begin,
-            text_callback=self._stt_text,
-            listenword_audio_callback=self._hotword_audio,
-            hotword_audio_callback=self._hotword_audio,
-            stopword_audio_callback=self._hotword_audio,
-            wakeupword_audio_callback=self._hotword_audio,
-            stt_audio_callback=self._stt_audio,
-            recording_audio_callback=self._recording_audio,
-        )
+
+        self._load_lock = RLock()
+        self._applied_config_hash = None
+        listener = self.config["listener"]
+        with self._load_lock:
+            self.voice_loop = self._init_voice_loop(listener)
+
+    def _config_hash(self):
+        config = {
+            "listener": self.config.get('listener'),
+            "hotwords": self.config.get('hotwords'),
+            "stt": self.config.get('stt'),
+            "opt_in": self.config.get('opt_in'),
+            "vad": self.config.get('VAD'),
+            "microphone": self.config.get('microphone')
+        }
+        return hash(json.dumps(config))
+
+    def _init_voice_loop(self, listener_config: dict):
+        """
+        Initialize a DinkumVoiceLoop object with the specified config
+        @param listener_config:
+        @return: Initialized VoiceLoop object
+        """
+        with self._load_lock:
+            self._applied_config_hash = self._config_hash()
+            loop = DinkumVoiceLoop(
+                mic=self.mic,
+                hotwords=self.hotwords,
+                stt=self.stt,
+                fallback_stt=self.fallback_stt,
+                vad=self.vad,
+                transformers=self.transformers,
+                #
+                speech_seconds=listener_config.get("speech_begin", 0.3),
+                silence_seconds=listener_config.get("silence_end", 0.7),
+                timeout_seconds=listener_config.get("recording_timeout", 10),
+                num_stt_rewind_chunks=listener_config.get(
+                    "utterance_chunks_to_rewind", 2),
+                num_hotword_keep_chunks=listener_config.get(
+                    "wakeword_chunks_to_save", 15),
```

Comment:
these should be added to readme

Reply:
There is currently no configuration documentation in the readme; can that go in a separate PR? I'm not familiar with all of the configuration options here